src/main/java/org/killbill/billing/plugin/adyen/client/payment/service/BaseAdyenPaymentServiceProviderPort.java

```diff
@@ -20,33 +20,130 @@
-                         .append("'");
+        logBuffer.append(", error=false");
+
+        logger.info(logBuffer.toString());
+    }
+
+    protected void logTransaction(final String transactionType, final String pspReference, String merchantAccount, final PaymentData paymentData, final PaymentModificationResponse response, final AdyenCallResult<?> adyenCall) {
+        final StringBuilder logBuffer = new StringBuilder();
+        appendTransactionType(logBuffer, transactionType);
+        appendMerchantAccount(logBuffer, merchantAccount);
+        appendPaymentData(logBuffer, paymentData);
+        appendPspReference(logBuffer, pspReference);
+        appendModificationResponse(logBuffer, response);
+        appendDuration(logBuffer, adyenCall.getDuration());
+        logBuffer.append(", error=false");
+
+        logger.info(logBuffer.toString());
+    }
+
+    protected void logTransactionError(final String transactionType, final UserData userData, String merchantAccount, final PaymentData paymentData, final AdyenCallResult<?> adyenCall) {
+        final StringBuilder logBuffer = new StringBuilder();
+        appendTransactionType(logBuffer, transactionType);
+        appendMerchantAccount(logBuffer, merchantAccount);
+        appendPaymentData(logBuffer, paymentData);
+        appendUserData(logBuffer, userData);
+        appendAdyenCall(logBuffer, adyenCall);
+        logBuffer.append(", error=true");
+
+        logger.info(logBuffer.toString());
+    }
+
+    protected void logTransactionError(final String transactionType, final String pspReference, String merchantAccount, final PaymentData paymentData, final AdyenCallResult<?> adyenCall) {
+        final StringBuilder logBuffer = new StringBuilder();
+        appendTransactionType(logBuffer, transactionType);
+        appendMerchantAccount(logBuffer, merchantAccount);
+        appendPspReference(logBuffer, pspReference);
+        appendPaymentData(logBuffer, paymentData);
+        appendPspReference(logBuffer, pspReference);
+        appendAdyenCall(logBuffer, adyenCall);
+        logBuffer.append(", error=true");
+
+        logger.info(logBuffer.toString());
+    }
+
+    private void appendTransactionType(final StringBuilder buffer, final String transactionType) {
+        buffer.append("op='").append(transactionType).append("'");
+    }
+
+    private void appendMerchantAccount(final StringBuilder buffer, final String merchantAccount) {
+        buffer.append(", merchantAccount='").append(merchantAccount).append("'");
+    }
+
+    private void appendPaymentData(final StringBuilder buffer, final PaymentData paymentData) {
+        if (paymentData == null ) {
+            return;
         }
+
+        if (paymentData.getAmount() != null) {
+            buffer.append(", amount='")
+                  .append(paymentData.getAmount())
+                  .append("'");
+        }
+        if (paymentData.getAmount() != null) {
```

Comment:
Actually, this was a copy/paste duplication error, but sure adding the currency is a good idea 😄